### DISCUSSION TITLE
Testing Interception object itself in Cypress

### DISCUSSION THREAD

**Discussion by Eprince-hub** [cypress-io/cypress#28220] [Questions and Help]
I am setting up a Cypress test for CSP violations, and I want to confirm the method I am using to ensure I am not doing anything crazy!

Most of the interception checks I have seen on the internet (Blog posts, Documentation) have been testing the request or the response body or other properties of the interception, like the example below.
```
cy.wait('@apiCheck').then((interception) => {
  assert.isNotNull(interception.response.body,
})
```

In my case, though, I don't care about the response or the request body because I just want to ensure no CSP violations. So whenever there is no CSP violation, the `interception` returns null (`logCSP Violations: null`). Only when there is a CSP violation does the interception return the request body in my case, but in this case, I want to throw an error immediately. 

So this is how my interception is set up.

```ts
cy.get('@cspAttacks').then((interception) => {
  cy.log('CSP Violations:', interception);
  expect(interception).to.equal(
    null,
    'Expected no CSP violations, but found one,'
  );
});
```
I am trying to make documentation for this approach with some documented proof that I am not doing anything wrong here, but I can't find anything.

Is this my approach okay, and would it not be flaky? And if not, what could be the best way to make this kind of check in Cypress?

Is there any example of a similar use case that I can refer to in my documentation?

I will be happy for any information please.

**Comment by portyu9** [comment]
Hi @Eprince-hub 

I would not use `interception === null` as the main way to prove that no CSP violation occurred.

Cypress now provides a clearer API for inspecting all requests captured by an intercept:

```js
cy.intercept('POST', '**/your-csp-report-endpoint**').as('cspReports')

// perform the complete flow that could generate CSP reports

cy.get('@cspReports.all').should('have.length', 0)
```

`@cspReports.all` gives you the complete set of interceptions Cypress has observed for that alias.

There is one important timing consideration: an assertion for a length of `0` can pass immediately. A test can only prove that **no report occurred before the point where you check it**; it cannot prove that a future asynchronous request will never occur.

For CSP violations, an even stronger pattern can be to fail immediately whenever the report endpoint is called:

```js
cy.intercept('POST', '**/your-csp-report-endpoint**', (req) => {
  throw new Error(
    `Unexpected CSP violation: ${JSON.stringify(req.body)}`
  )
})
```

Register that intercept **before** loading/exercising the page and keep the test running through the complete behavior you want to cover.

You can also use `.all` afterwards for diagnostics:

```js
cy.get('@cspReports.all').then((reports) => {
  cy.log(`CSP reports: ${reports.length}`)
})
```

So the assertion itself is straightforward; the important part for avoiding flake is choosing a deterministic end point in your application flow before declaring that no CSP reports occurred.

Please mark this answer as accepted if it helped, thank you.
